internal/services/netapp/netapp_account_encryption_resource.go

```diff
@@ -294,17 +321,52 @@ func expandEncryption(ctx context.Context, input string, keyVaultsClient *keyVau
 
 	keyId, err := keyVaultParse.ParseOptionallyVersionedNestedKeyID(input)
 	if err != nil {
-		return nil, fmt.Errorf("parsing `key_vault_key_id`: %+v", err)
+		return nil, fmt.Errorf("parsing `key_vault_key_id` %q: %+v", input, err)
 	}
 
-	keyVaultID, err := keyVaultsClient.KeyVaultIDFromBaseUrl(ctx, subscriptionID, keyId.KeyVaultBaseUrl)
-	if err != nil {
-		return nil, fmt.Errorf("retrieving the resource id the key vault at url %q: %s", keyId.KeyVaultBaseUrl, err)
-	}
+	var keyVaultResourceID string
+
+	// For cross-tenant scenarios, we can't lookup the key vault from the current tenant
+	// Instead, we need to construct the resource ID manually or use a provided resource ID
+	if model.FederatedClientID != "" {
+		// Cross-tenant scenario: try to construct the key vault resource ID from the URL
+		// Extract the key vault name from the URL for all Azure cloud environments
+		keyVaultName := ""
+		if keyId.KeyVaultBaseUrl != "" {
+			const httpsPrefix = "https://"
+			if strings.HasPrefix(keyId.KeyVaultBaseUrl, httpsPrefix) {
+				// Remove the "https://" prefix
+				urlWithoutProtocol := keyId.KeyVaultBaseUrl[len(httpsPrefix):]
```

Comment:
I believe we can shorten a lot of this down with `urlWithoutProtocol = strings.TrimPrefix(keyId.KeyVaultBaseUrl, "https://")`